src/main/java/lotto/domain/LottoMachine.java

```diff
@@ -1,59 +1,100 @@
 package lotto.domain;
 
-import lotto.constant.LottoConstant;
 import lotto.constant.Prize;
 
-
-import java.text.DecimalFormat;
 import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collections;
 import java.util.EnumMap;
+import java.util.HashSet;
 import java.util.List;
 import java.util.Set;
 import java.util.stream.Collectors;
 
-public abstract class LottoMachine {
+public class LottoMachine {
 
     private final int cashAmount;
-    private final List<Integer> balls = new ArrayList<>();
+    private final LottoNumbers balls;
+    private ProfitRateStrategy profitRateStrategy;
+    private static final int LOTTO_PRICE = 1000;
+    public static final int LOTTO_SIZE = 6;
 
-    public LottoMachine(int cashAmount) {
-        if (cashAmount < LottoConstant.LOTTO_PRICE) {
-            throw new IllegalArgumentException("로또 구입 금액은 최소 " + LottoConstant.LOTTO_PRICE + "원 이상이어야 합니다.");
-        }
+    private LottoMachine(int cashAmount) {
         this.cashAmount = cashAmount;
-        for (int i = LottoConstant.LOTTO_START_NUMBER; i <= LottoConstant.LOTTO_END_NUMBER; i++) {
-            balls.add(i);
+        this.balls = new LottoNumbers();
+        this.profitRateStrategy = new MissionProfitRateStrategy();
+    }
+
+    public void setProfitRateStrategy(ProfitRateStrategy profitRateStrategy) {
+        this.profitRateStrategy = profitRateStrategy;
+    }
+
+    public static LottoMachine of(int cashAmount) {
+        validateCashAmount(cashAmount);
+        return new LottoMachine(cashAmount);
+    }
+
+    private static void validateCashAmount(int cashAmount) {
+        if (cashAmount < LOTTO_PRICE) {
+            throw new IllegalArgumentException("로또 구입 금액은 최소 " + LOTTO_PRICE + "원 이상이어야 합니다.");
         }
     }
 
-    public List<Integer> getBalls() {
-        return Collections.unmodifiableList(balls);
+    public List<Lotto> createAutomatically() {
+        List<Lotto> lottoList = new ArrayList<>();
+        int purchaseCount = calculatePurchaseCount();
+        for (int i = 0; i < purchaseCount; i++) {
+            balls.shuffle();
+            Set<Integer> lottoBalls = new HashSet<>(balls.getNumbers().subList(0, LOTTO_SIZE));
+            lottoList.add(new Lotto(lottoBalls));
+        }
+        return lottoList;
```

Comment:
자동 로또를 만드는 방식을 추상화해보는건 어떨까요? 그럼 다음과 같이도 가능할 것 같네요! 

```suggestion
    public List<Lotto> createAutomatically() {
        int purchaseableQuantity = calculatePurchaseableQuantity();
        if (purchaseableQuantity < 1) { ... }
        
        return IntStream.range(0, purchaseableQuantity)
                         .map(idx-> lottoCreateStrategy.create())
                         .collect(Collectors.toList());
```